src/hera/workflows/runner.py

```diff
@@ -122,7 +123,62 @@ def _map_keys(function: Callable, kwargs: dict) -> dict:
     return mapped_kwargs
 
 
-def _runner(entrypoint: str, kwargs_list: Any) -> str:
+def _save_outputs(
+    function_results: Union[Tuple[Any], Any], output_destinations: List[Tuple[type, Union[Parameter, Artifact]]]
+) -> None:
+    """Save the results of the function to the specified output destinations.
+
+    The results are matched with the specified outputs and saved using the schema:
+
+    If the artifact path is specified, that is used instead.
+    """
+    if not isinstance(function_results, tuple):
+        function_results = [function_results]
+    if len(function_results) != len(output_destinations):
+        raise ValueError("The number of outputs does not match the annotation")
+
+    outputs_dir = Path(RunnerScriptConstructor().outputs_directory)
```

Comment:
I'm not sure this will work in the context of the script runner in a user's image - they would have to call `set_class_defaults` at some point in their code (as in, within the executing function as opposed to in a config somewhere), otherwise they'll get the default `hera/outputs`